app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -75,7 +75,20 @@ def content_and_puppet_match?
       end
 
       def inherited_attributes_with_katello
-        self.content_facet.kickstart_repository_id ||= hostgroup.inherited_kickstart_repository_id if content_facet.present?
+        if hostgroup.present?
+          if content_facet.present?
+            self.content_facet.kickstart_repository_id ||= hostgroup.inherited_kickstart_repository_id
+          else
+            # facent may not be present when created via autoprovisioning
+            if hostgroup.inherited_content_view_id && hostgroup.inherited_lifecycle_environment_id
+              build_content_facet(
+                :kickstart_repository_id => hostgroup.inherited_kickstart_repository_id,
+                :content_source_id => hostgroup.inherited_content_source_id,
```

Comment:
Deleting this line, and removing 'content_view_id' and 'lifecycle_environment_id'  from the line below:
```
inherited_attributes_without_katello.concat(%w(content_source_id content_view_id lifecycle_environment_id))
```

Seems to make it all work.